UserSessionsMeter.c

```diff
@@ -0,0 +1,162 @@
+/*
+htop - UserSessionsMeter.c
+(C) 2024 htop dev team
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "config.h" // IWYU pragma: keep
+
+#include "UserSessionsMeter.h"
+
+#include <dlfcn.h>
+#include <utmpx.h>
+
+
+#include "CRT.h"
+#include "Macros.h"
+#include "Object.h"
+#include "RichString.h"
+#include "XUtils.h"
+
+#if defined(BUILD_STATIC) && defined(HAVE_LIBSYSTEMD)
+#include <systemd/sd-login.h>
+#endif
+
+
+static unsigned int users = 0;
+
+#ifdef BUILD_STATIC
+#define sym_sd_get_sessions sd_get_sessions
+#else
+
+static int (*sym_sd_get_sessions)(char***);
+static void* dlopenHandle = NULL;
+
+#endif /* BUILD_STATIC */
+
+
+static void UserSessionsMeter_done(ATTR_UNUSED Meter* this) {
+#ifndef BUILD_STATIC
+   if (dlopenHandle) {
+      dlclose(dlopenHandle);
+      dlopenHandle = NULL;
+   }
+#endif /* !BUILD_STATIC */
```

Comment:
Make the unload code a separate function, so that you won't need to duplicate the code for destroying `dlopenHandle`

```c
static void closeDlHandle(void) {
#ifndef BUILD_STATIC
   if (dlopenHandle) {
      dlclose(dlopenHandle);
      dlopenHandle = NULL;
   }
#else
   return; // No-op
#endif /* !BUILD_STATIC */
}

Reply:
Sorry. I take this suggestion back. It would be way better if the handle, `dlopen` and `dlclose` code be shared with the SystemdMeter. But that's a code refactoring to be done...

Reply:
That's also why I suggested to move the actual reading of values into the platform code. Systemd is Linux-specific after all and each platform needs things to be done slightly different anyway; though the utmp stuff can mostly be shared as `generic` code, as is e.g. already done with the uname stuff for the System meter to read OS/kernel info.